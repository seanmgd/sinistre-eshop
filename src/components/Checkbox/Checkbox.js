import PropTypes from 'prop-types'
import { StyledCheckbox } from './Checkbox.styles'

export function Checkbox({ disabled, checked }) {
  return (
    <StyledCheckbox
      disabled={disabled}
      defaultChecked={checked}
      type="checkbox"
      data-testid="checkbox"
    />
  )
}

Checkbox.propTypes = {
  disabled: PropTypes.bool,
  defaultChecked: PropTypes.bool,
}

Checkbox.defaultProps = {
  disabled: false,
  checked: false,
}
