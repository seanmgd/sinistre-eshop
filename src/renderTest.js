import { render as renderTest } from '@testing-library/react'
import { AppProviders } from './AppProviders'
import theme from './styles/theme'

// Export it first to allow overrides.
export * from '@testing-library/react'
export { renderInAppProviders as render }

export function renderInAppProviders(node, options) {
  const { rerender, ...res } = renderTest(
    <AppProviders theme={theme} children={node} />,
    options,
  )

  return {
    ...res,
    rerender: node => rerender(<AppProviders theme={theme} children={node} />),
  }
}
