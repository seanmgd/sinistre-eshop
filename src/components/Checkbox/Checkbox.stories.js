import React from 'react'
import { Checkbox } from './Checkbox'
import { boolean } from '@storybook/addon-knobs'
import { StyledCheckbox } from './Checkbox.styles'

export default {
  title: 'Components/Checkbox',
  component: Checkbox,
}

export const ToStorybook = () => (
  <StyledCheckbox
    disabled={boolean('disabled', Checkbox.defaultProps.disabled)}
    checked={boolean('checked', Checkbox.defaultProps.checked)}
  />
)

ToStorybook.story = {
  name: 'Checkbox',
}

export const Example = () => {
const [check, setCheck] = React.UseEffect
 return (
  <StyledCheckbox
      disabled={boolean('disabled', Checkbox.defaultProps.disabled)}
      checked={boolean('checked', Checkbox.defaultProps.checked)}
  />)
}


