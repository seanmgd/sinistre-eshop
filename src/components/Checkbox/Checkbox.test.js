import React from 'react'
import { render, fireEvent } from '../../renderTest'
import { Checkbox } from './Checkbox'

describe('<Checkbox />', () => {
  it('should render a unchecked Checkbox', () => {
    const { getByTestId } = render(<Checkbox />)
    const checkbox = getByTestId('checkbox')

    expect(checkbox.checked).toEqual(false)
  })
  it('should render a checked Checkbox', () => {
    const { getByTestId } = render(<Checkbox />)
    const checkbox = getByTestId('checkbox')

    fireEvent.click(checkbox)
    expect(checkbox.checked).toEqual(true)
  })
  it('should render a undisabled Checkbox', () => {
    const { getByTestId } = render(<Checkbox />)
    const checkbox = getByTestId('checkbox')

    expect(checkbox.disabled).toEqual(false)
  })
  it('should render a disabled Checkbox', () => {
    const { getByTestId } = render(<Checkbox disabled />)
    const checkbox = getByTestId('checkbox')

    expect(checkbox.disabled).toEqual(true)
  })
})
