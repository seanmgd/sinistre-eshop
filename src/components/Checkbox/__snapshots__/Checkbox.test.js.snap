// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render a checked Checkbox 1`] = `
<input
  class="sc-bdfBwQ gTBYiw"
  data-testid="checkbox"
  type="checkbox"
/>
`;

exports[`<Checkbox /> should render a disabled Checkbox 1`] = `
<input
  class="sc-bdfBwQ gTBYiw"
  data-testid="checkbox"
  disabled=""
  readonly=""
  type="checkbox"
/>
`;

exports[`<Checkbox /> should render a unchecked Checkbox 1`] = `
<input
  class="sc-bdfBwQ gTBYiw"
  data-testid="checkbox"
  readonly=""
  type="checkbox"
/>
`;

exports[`<Checkbox /> should render a undisabled Checkbox 1`] = `
<input
  class="sc-bdfBwQ gTBYiw"
  data-testid="checkbox"
  readonly=""
  type="checkbox"
/>
`;
